flexget/plugins/filter/upgrade.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import unicode_literals, division, absolute_import
+from builtins import *
+from collections import defaultdict
+from datetime import datetime
+from sqlalchemy import Column, String, Unicode, DateTime, Boolean
+import logging
+from os.path import expanduser
+
+from flexget import db_schema, plugin
+from flexget.entry import Entry
+from flexget.utils.database import quality_property
+from flexget.db_schema import Session
+from flexget.event import event
+
+log = logging.getLogger('upgrade')
+
+Base = db_schema.versioned_base('upgrade', 0)
+
+entry_actions = {
+    'accept': Entry.accept,
+    'reject': Entry.reject,
+    'fail': Entry.fail
+}
+
+
+class EntryUpgrade(Base):
+    __tablename__ = 'upgrade'
+
+    id = Column(Unicode, primary_key=True, index=True)
+    title = Column(Unicode)
+    _quality = Column('quality', String)
+    quality = quality_property('_quality')
+    proper = Column(Boolean)
+    added = Column(DateTime, index=True)
+
+    def __init__(self):
+        self.added = datetime.now()
+
+    def __str__(self):
+        return '<Upgrade(id=%s,added=%s,quality=%s)>' % \
+               (self.id, self.added, self.quality)
+
+
+def group_entries(entries, identified_by):
+    grouped_entries = defaultdict(list)
+
+    # Group by Identifier
+    for entry in entries:
+        identifier = entry.get('id') if identified_by == 'auto' else entry.render(identified_by)
+        if identifier:
```

Comment:
Just my personal preference, I prefer to keep the flow control leaner:
```python
if not identifier:
    log.debug('No identifier found for', entry['title'])
    continue
grouped_entries[identifier.lower()].append(entry)
``` 
just a suggestion obviously

Reply:
changed